server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileCompareAction.java

```diff
@@ -0,0 +1,205 @@
+      .setSince("5.2");
+
+    compare.createParam(PARAM_LEFT_KEY)
+      .setDescription("A profile key.")
+      .setExampleValue("java-sonar-way-12345")
+      .setRequired(true);
+
+    compare.createParam(PARAM_RIGHT_KEY)
+      .setDescription("Another profile key.")
+      .setExampleValue("java-sonar-way-with-findbugs-23456")
+      .setRequired(true);
+  }
+
+  @Override
+  public void handle(Request request, Response response) throws Exception {
+    String leftKey = request.mandatoryParam(PARAM_LEFT_KEY);
+    String rightKey = request.mandatoryParam(PARAM_RIGHT_KEY);
+
+    QProfileComparisonResult result = comparator.compare(leftKey, rightKey);
+
+    List<Rule> referencedRules = ruleService.getByKeys(result.collectRuleKeys());
+    Map<RuleKey, Rule> rulesByKey = Maps.uniqueIndex(referencedRules, new NonNullInputFunction<Rule, RuleKey>() {
+      @Override
+      protected RuleKey doApply(Rule input) {
+        return input.key();
+      }
+    });
+
+    writeResult(response.newJsonWriter(), result, rulesByKey);
+  }
+
+  private void writeResult(JsonWriter json, QProfileComparisonResult result, Map<RuleKey, Rule> rulesByKey) {
+    json.beginObject();
+
+    json.name("left").beginObject();
+    writeProfile(json, result.left());
+    json.endObject();
+
+    json.name("right").beginObject();
+    writeProfile(json, result.right());
+    json.endObject();
+
+    json.name("inLeft");
+    writeRules(json, result.inLeft(), rulesByKey);
+
+    json.name("inRight");
+    writeRules(json, result.inRight(), rulesByKey);
+
+    json.name("modified");
+    writeDifferences(json, result.modified(), rulesByKey);
+
+    json.name("same");
+    writeRules(json, result.same(), rulesByKey);
+
+    json.endObject().close();
+  }
+
+  private void writeProfile(JsonWriter json, QualityProfileDto profile) {
+    json.prop("key", profile.getKey())
+      .prop("name", profile.getName());
+  }
+
+  private void writeRules(JsonWriter json, Map<RuleKey, ActiveRule> activeRules, Map<RuleKey, Rule> rulesByKey) {
+    json.beginArray();
+    for (Entry<RuleKey, ActiveRule> activeRule : activeRules.entrySet()) {
+      RuleKey key = activeRule.getKey();
+      ActiveRule value = activeRule.getValue();
+
+      json.beginObject();
+      writeRule(json, key, rulesByKey);
+      json.prop("severity", value.severity());
+      json.endObject();
+    }
+    json.endArray();
+  }
+
+  private void writeRule(JsonWriter json, RuleKey key, Map<RuleKey, Rule> rulesByKey) {
+    String repositoryKey = key.repository();
+    json.prop("key", key.toString())
+      .prop("name", rulesByKey.get(key).name())
+      .prop("pluginKey", repositoryKey);
+
+    Repository repo = ruleRepositories.repository(repositoryKey);
+    if (repo != null) {
+      String languageKey = repo.getLanguage();
+      Language language = languages.get(languageKey);
+
+      json.prop("pluginName", repo.getName());
+      json.prop("languageKey", languageKey);
```

Comment:
I'm wondering, does it make sense to compare 2 profiles from different language ? I know that in rails it was possible, but I'm use it makes sense to keep this feature...